src/v/cluster/archival_metadata_stm.cc

```diff
@@ -1015,81 +942,94 @@ ss::future<> archival_metadata_stm::apply(const model::record_batch& b) {
               }
           });
     } else {
-        b.for_each_record([this,
-                           base_offset = b.base_offset()](model::record&& r) {
-            auto key = serde::from_iobuf<cmd_key>(r.release_key());
-
-            if (key != mark_clean_cmd::key) {
-                // All keys other than mark clean make the manifest dirty
-                _last_dirty_at = base_offset + model::offset{r.offset_delta()};
-            }
+        auto on_exit = ss::defer(
```

Comment:
Is the STM re-created on leadership change?

Reply:
It's not re-created.

Reply:
Follow up question in wrong thread: https://github.com/redpanda-data/redpanda/pull/16774#discussion_r1532573437